### PR TITLE
ci: generate apidocs when pushing to default branch

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Configure CI to generate rust api docs and push to gh-pages.

This patch also updates our toolchain to fix an ICE when running `cargo doc`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: .github/workflows/apidoc.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  apidoc:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: ${{ env.RUST_TOOLCHAIN }}
+    - run: cargo doc
```
Comment (waynexia) [inline .github/workflows/apidoc.yml:26]:
```suggestion
    - run: cargo doc --workspace --no-deps --document-private-items
```

These flags: ignore deps, and only generate documents for all items in our workspace.

Comment (sunng87) [reply .github/workflows/apidoc.yml:26]: hmm.. I was adding deps doc intentionally so we can find doc for all APIs available.

Comment (sunng87) [reply .github/workflows/apidoc.yml]: exclude deps to speedup

Comment (waynexia) [reply .github/workflows/apidoc.yml]: Docs of our dependency are available in `docs.rs`, we may not need to replicate one.

**Review by waynexia — APPROVED** [review]
👍

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**